src/app.js

```diff
@@ -39,7 +38,7 @@ export default function (app) {
 
     if (obj = plugin.hooks) {
       Object.keys(obj).forEach(function (key) {
-        hooks[key].push(obj[key])
+        (hooks[key] ? hooks[key] : hooks[key] = []).push(obj[key])
```

Comment:
This suggests to me that hooks from plugins get aggregated into an array of _callbacks_ for that type of hook.. this is not specific to the `onRoute` hook but all hooks right?
 
 If that assumption is true then the router would need to do something like..

```
app.hooks.onRoute.forEach(hook => hook(model))
```
Right?